src/worleynoise.rs

```diff
@@ -41,189 +56,198 @@ fn worley_noise(
+impl NoiseCellMap {
+    fn new(reg_size: i32, reg_amt: i32) -> Self {
+        let mut noise_cell_map = NoiseCellMap {
+            reg_vec: Vec::new(),
+            reg_size,
+            reg_amt,
         };
-
-        map.init_regions();
-
-        for x in 0..cell_map_width {
-            map.cell_map.push(Vec::new());
-            for y in 0..cell_map_height {
-                let cell = Cell::new(
-                    x,
-                    y,
-                    map.region_map[(x / region_size) as usize][(y / region_size) as usize].clone(),
-                );
-                map.cell_map[(x) as usize].push(cell);
+        for x in 0..reg_amt {
+            noise_cell_map.reg_vec.push(Vec::new());
+            for y in 0..reg_amt {
+                noise_cell_map.reg_vec[x as usize].push(NoiseCellRegion::new((x, y), reg_size));
             }
         }
-        map
+        noise_cell_map
     }
-    fn init_regions(&mut self) {
-        let mut rng = rand::thread_rng();
-
-        let regions_x = self.cell_map_width / self.region_size;
-        let regions_y = self.cell_map_height / self.region_size;
-        //those two variables ensure that we dont EVER panic due to not having enough nodes spawned for the distance algorithm to function.
-        let mut node_count = 0;
-        let mut distance_in_regions_since_last_node = 0;
-
-        for i in 0..regions_x {
-            distance_in_regions_since_last_node += 1;
-            self.region_map.push(Vec::new());
-            for j in 0..regions_y {
-                distance_in_regions_since_last_node += 1;
-                let mut region = Region::new(i, j);
-                if rng.gen_range(0..100) as f32 <= self.node_chance || node_count < 2 {
-                    let xcord = rng.gen_range(0..self.region_size);
-                    let ycord = rng.gen_range(0..self.region_size);
-                    let node =
-                        Node::new(xcord + i * self.region_size, ycord + j * self.region_size);
-                    region.node = Some(node);
-                    node_count += 1;
-                    distance_in_regions_since_last_node = 0;
-                }
 
-                if distance_in_regions_since_last_node > 3 {
-                    node_count = 0;
+    fn node_fill(&mut self, mut node_min: u32, mut node_max: u32, node_chance: usize) -> &mut Self {
+        node_min = node_min.max(1);
+        node_max = node_min.max(node_max);
+        let node_counter: Arc<RwLock<usize>> = Arc::new(RwLock::new(0));
+        let reg_size = self.reg_size;
+        self.reg_vec.par_iter_mut().flatten().for_each(|region| {
+            let mut rng = ThreadRng::default();
+            {
+                let mut write_guard = node_counter.write().unwrap();
+                if (*write_guard < RANGE) && rng.gen_range(0..100) > node_chance {
+                    *write_guard += 1;
+                    return;
                 }
+                *write_guard = 0;
+            }
```

Comment:
what's even going on here, comment?

Reply:
basically, to make sure the algorithm works optimally we **have to** ensure we spawn at least some nodes, this amount of nodes scales inversely to range.

Reply:
like could you put that in a `// comment`